NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -8,9 +10,18 @@ CategoryProduct(c::CategoryProduct) = _CategoryProduct(categories(c))
 
 categories(s::CategoryProduct) = s.cats
 
-Base.isempty(S::CategoryProduct) = isempty(categories(S))
-Base.length(S::CategoryProduct) = length(categories(S))
-Base.getindex(S::CategoryProduct, args...) = getindex(categories(S), args...)
+Base.isempty(s::CategoryProduct) = isempty(categories(s))
+Base.length(s::CategoryProduct) = length(categories(s))
+Base.getindex(s::CategoryProduct, args...) = getindex(categories(s), args...)
```

Comment:
How often are these used? I wonder if these are necessary.

In the cases where they are used, we could just use `isempty(categories(s))`, `length(categories(s))`, etc.

My reason for wanting to be careful about these definitions is that in general I think a `CategoryProduct` should act more like an `AbstractCategory` rather than a collection of categories. None of these definitions make sense for other `AbstractCategory` subtypes.

Reply:
@emstoudenmire probably you added these definitions so it would be good to get your feedback as well.

Reply:
I have no strong opinion here. Maybe indeed removing this function would be better.

Reply:
Yes let's be conservative here. Both `isempty` and `length` can certainly go. The test file will just have to be modified because it uses `length` but actually the implementation does not (I just checked).

I found `getindex` convenient to use for the test, but it is also just a pass-through to `categories` so one can always write the tests like:
```julia
s = sector("B" => Ising("ψ"), :C => Z{2}(1))
c = categories(s)
@test c[:B] == Ising("ψ")
```
So no problem to get rid of `getindex` too and just rewrite the tests in that style.

Reply:
Yeah, I would vote for removing these definitions and updating the tests accordingly.

Reply:
Ok I made this change, and I think I updated the tests correctly. However, the tests are failing for many other reasons due to changes in this PR, so I'll leave it up to @ogauthe to fix those.